lib/grit_adapter/git_layer_grit.rb

```diff
@@ -166,6 +166,12 @@ def rev_list(options, *refs)
       rescue Grit::GitRuby::Repository::NoSuchShaFound
         raise Gollum::Git::NoSuchShaFound
       end
+
+      def revert(path, sha1, sha2, ref)
+        patch = path ?
+          repo.diff(sha2, sha1, path).first.diff : repo.diff(sha2, sha1).map { |d| d.diff }.join("\n")
+        apply_patch(ref, patch)
```

Comment:
Given the brevity of the `apply_patch` implementation, it might just make sense to call `@git.apply_patch({}, ref, patch)` here. Unless of course the `apply_patch` method is something we want to have exposed in all of our adapters?